administrator/components/com_content/controllers/article.php

```diff
@@ -145,6 +146,30 @@ public function batch($model = null)
 	}
 
 	/**
+	 * Method to generate and store share token.
+	 *
+	 * @return  boolean   True if token successfully stored, false otherwise and internal error is set.
+	 *
+	 * @since   3.7
+	 */
+	public function shareDraft()
+	{
+		if (!JSession::checkToken('get'))
+		{
+			$this->app->enqueueMessage(JText::_('JINVALID_TOKEN'));
+			$this->app->redirect('index.php');
```

Comment:
You are inside an AJAX call, no need to do a redirect. You need this code:
```
$app->enqueueMessage(JText::_('JINVALID_TOKEN'), 'error');
			echo new JResponseJson;
			$app->close();
```
You also need to declare the $app otherwise you can't use it. $this->app doesn't exist.
You